src/modules/content-s3/s3.c

```diff
@@ -22,20 +22,16 @@
 
 #include "s3.h"
 
-static S3Status status_g;
 static S3Protocol protocol = S3ProtocolHTTP;
 static S3UriStyle uri_style = S3UriStylePath;
 
 /* Data needed by the get object callback function
  */
-struct get_cb_data {
+struct cb_data {
```

Comment:
Ah, OK you went with one callback struct for all the callbacks.  That seems to work out.  We may have to change it later if we ever add callback-specific members but I guess that can come later.  This is fairly neat and tidy.

One nit is that we don't typically use variable names that begin with underbars in the flux code base.  I see you needed to disambiguate between the generic callback arg and the struct pointer.  Maybe just call the struct pointer `cb_data` instead of `_data`?